src/uu/rm/src/rm.rs

```diff
@@ -396,8 +412,17 @@ fn handle_dir(path: &Path, options: &Options) -> bool {
     } else if options.dir && (!is_root || !options.preserve_root) {
         had_err = remove_dir(path, options).bitor(had_err);
     } else if options.recursive {
-        show_error!("could not remove directory {}", path.quote());
-        had_err = true;
+        if is_root || !options.preserve_root {
+            show_error!(
+                "it is dangerous to operate recursively on '{}'",
+                MAIN_SEPARATOR
+            );
+            show_error!("use --no-preserve-root to override this failsafe");
+            had_err = true;
+        } else {
+            show_error!("could not remove directory {}", path.quote());
+            had_err = true;
+        }
```

Comment:
When does this arm get reached? In order to get here, all of the following must hold:
- `options.recursive && (!is_root || !options.preserve_root) == false` (because the first `if` wasn't taken)
- (the second `if` wasn't taken, but that is trivially satisfiable)
- `options.recursive == true` (because this `if` was taken)
- `is_root || !options.preserve_root == false` because the sub-if wasn't taken

This simplifies down to `is_root && options.preserve_root == true` and `!is_root && options.preserve_root == true`, a contradiction. This piece of code is unreachable.

Is it possible that the condition `if is_root || !options.preserve_root {` is slightly mistaken? I'm not even sure which scenario was meant here, so I can't even make a reasonable suggestion.

Reply:
You are right, i was mistaken. For some reason i thought of ``!options.preserve_root`` as ``--preserve-root``.

I think we can just have the old condition then ``if options.recursive`` and then just change the error message to the correct one. We catch the non root directories in ``options.recursive && (!is_root || !options.preserve_root)`` so i think just changing it should be correct.